### PR TITLE
[SMALLFIX] adding support for HdfsFileInputStream.available()

### PR DESCRIPTION
**Pull request by jsimsa**
Addresses: https://groups.google.com/forum/#!topic/alluxio-users/7nbVt7aF8no

### CODE REVIEW

File: core/client/src/main/java/alluxio/hadoop/HdfsFileInputStream.java [hunk 1]
```diff
@@ -104,7 +104,7 @@ public HdfsFileInputStream(AlluxioURI uri, org.apache.hadoop.conf.Configuration
    */
   @Override
   public int available() throws IOException {
-    throw new IOException(ExceptionMessage.NOT_SUPPORTED.getMessage());
+    return (int) mAlluxioFileInputStream.remaining();
```
Comment (aaudiber) [inline core/client/src/main/java/alluxio/hadoop/HdfsFileInputStream.java]: update javadoc

File: tests/src/test/java/alluxio/hadoop/HdfsFileInputStreamIntegrationTest.java [hunk 2]
```diff
@@ -75,7 +75,16 @@ public final void before() throws IOException, AlluxioException {
   }
 
   /**
-   * Test {@link HdfsFileInputStream#read()}.
+   * Tests {@link HdfsFileInputStream#available()}.
+   */
+  @Test
+  public void availableTest() throws IOException {
```
Comment (aaudiber) [inline tests/src/test/java/alluxio/hadoop/HdfsFileInputStreamIntegrationTest.java]: advance the streams by some amount and then verify .available()

**Comment by aaudiber** [comment]
Thanks for adding this! I left a couple minor comments

File: tests/src/test/java/alluxio/hadoop/HdfsFileInputStreamIntegrationTest.java (hunk 2 above)
Comment (jsimsa) [inline tests/src/test/java/alluxio/hadoop/HdfsFileInputStreamIntegrationTest.java]: done

File: core/client/src/main/java/alluxio/hadoop/HdfsFileInputStream.java (hunk 1 above)
Comment (jsimsa) [inline core/client/src/main/java/alluxio/hadoop/HdfsFileInputStream.java]: done

**Comment by jsimsa** [comment]
@aaudiber addressed your comments, PTAL

**Comment by aaudiber** [comment]
LGTM

**Pull request merged**